commit d9756ce618f33ab0d2bf2983b49c32e2017c8b7b
Author: Kees Cook <kees@kernel.org>
Date:   Mon Oct 9 09:10:35 2023 -0700

    wifi: p54: Annotate struct p54_cal_database with __counted_by
    
    Prepare for the coming implementation by GCC and Clang of the __counted_by
    attribute. Flexible array members annotated with __counted_by can have
    their accesses bounds-checked at run-time via CONFIG_UBSAN_BOUNDS (for
    array indexing) and CONFIG_FORTIFY_SOURCE (for strcpy/memcpy-family
    functions).
    
    Add __counted_by for struct p54_cal_database.
    
    Cc: Christian Lamparter <chunkeey@googlemail.com>
    Cc: Kalle Valo <kvalo@kernel.org>
    Cc: "Gustavo A. R. Silva" <gustavoars@kernel.org>
    Cc: linux-wireless@vger.kernel.org
    Cc: linux-hardening@vger.kernel.org
    Suggested-by: Jason Andryuk <jandryuk@gmail.com>
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Reviewed-by: Jason Andryuk <jandryuk@gmail.com>
    Signed-off-by: Kalle Valo <kvalo@kernel.org>
    Link: https://lore.kernel.org/r/20231009161028.it.544-kees@kernel.org

diff --git a/drivers/net/wireless/intersil/p54/p54.h b/drivers/net/wireless/intersil/p54/p54.h
index 3356ea708d81..522656de4159 100644
--- a/drivers/net/wireless/intersil/p54/p54.h
+++ b/drivers/net/wireless/intersil/p54/p54.h
@@ -126,7 +126,7 @@ struct p54_cal_database {
 	size_t entry_size;
 	size_t offset;
 	size_t len;
-	u8 data[];
+	u8 data[] __counted_by(len);
 };
 
 #define EEPROM_READBACK_LEN 0x3fc